src/androidTest/java/com/segment/analytics/android/integrations/appboy/AppboyIntegrationOptionsAndroidTest.java

```diff
@@ -2,50 +2,131 @@
+  private static final String TRAIT_EMAIL_UPDATED = "updated@segment.com";
+  private static final String TRAIT_CITY = "city";
+  private static final String TRAIT_COUNTRY = "country";
+  private static final String CUSTOM_TRAIT_KEY = "custom_trait_key";
+  private static final String CUSTOM_KEY_VALUE = "custom_key_value";
+
+  @Mock Appboy appboy;
+  @Mock AppboyUser appboyUser;
+
+  private AppboyIntegration appboyIntegration;
+  private PreferencesTraitsCache traitsCache;
 
   @BeforeClass
   public static void beforeClass() {
     AppboyConfig appboyConfig = new AppboyConfig.Builder().setApiKey("testkey").build();
     Appboy.configure(getContext(), appboyConfig);
   }
 
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
+
+    when(appboy.getCurrentUser()).thenReturn(appboyUser);
+
+    Logger logger = Logger.with(Analytics.LogLevel.DEBUG);
+
+    traitsCache = new PreferencesTraitsCache(getContext());
+
+    appboyIntegration = new AppboyIntegration(
+        appboy, "foo", logger, true,
+        traitsCache, new ReplaceUserIdMapper());
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    traitsCache.clear();
+  }
+
   @Test
   public void testUserIdMapperTransformsAppboyUserId() {
-    Traits traits = createTraits(ORIGINAL_USER_ID);
-    IdentifyPayload identifyPayload = new IdentifyPayloadBuilder().traits(traits).build();
+    Traits traits = createTraits(USER_ID);
 
-    Logger logger = Logger.with(Analytics.LogLevel.DEBUG);
+    callIdentifyWithTraits(traits);
 
-    AppboyIntegration integration = new AppboyIntegration(Appboy.getInstance(getContext()), "foo", logger, true, new ReplaceUserIdMapper());
+    verify(appboy).changeUser(TRANSFORMED_USER_ID);
+  }
 
-    integration.identify(identifyPayload);
+  @Test
+  public void testShouldNotTriggerAppboyUpdateIfTraitDoesntChange() {
+    Traits traits = createTraits(USER_ID);
+
+    Traits.Address address = new Traits.Address();
+    address.putCity(TRAIT_CITY);
+    address.putCountry(TRAIT_COUNTRY);
+
+    traits.putAddress(address);
+    traits.putEmail(TRAIT_EMAIL);
+    traits.put(CUSTOM_TRAIT_KEY, CUSTOM_KEY_VALUE);
+
+    callIdentifyWithTraits(traits);
+    callIdentifyWithTraits(traits);
+    callIdentifyWithTraits(traits);
+
+    verify(appboyUser, times(1)).setEmail(TRAIT_EMAIL);
+    verify(appboyUser, times(1)).setHomeCity(TRAIT_CITY);
+    verify(appboyUser, times(1)).setCountry(TRAIT_COUNTRY);
+    verify(appboyUser, times(1)).setCustomUserAttribute(CUSTOM_TRAIT_KEY, CUSTOM_KEY_VALUE);
+  }
+
+  @Test
+  public void testShouldTriggerUpdateIfTraitChanges() {
+    Traits traits = createTraits(USER_ID);
+    traits.putEmail(TRAIT_EMAIL);
+    callIdentifyWithTraits(traits);
+    callIdentifyWithTraits(traits);
+
+    Traits traitsUpdate = createTraits(USER_ID);
+    traitsUpdate.putEmail(TRAIT_EMAIL_UPDATED);
+    callIdentifyWithTraits(traitsUpdate);
+    callIdentifyWithTraits(traitsUpdate);
```

Comment:
You could skip the duplicated call in this test. You're already checking that feature in the previous test, so there's no need to make this test fail also. But no big deal really 🤷‍♂